src/fshtypes/Configuration.ts

```diff
@@ -139,6 +139,10 @@ export type Configuration = {
   // The instanceOptions property is used to configure certain aspects of how SUSHI processed instances.
   // See the individual option definitions in ConfigurationInstanceOptions for more detail.
   instanceOptions?: ConfigurationInstanceOptions;
+
+  // When set to true, slices must be referred to by name and not a numeric index in order to be used
+  // in an Instance's assignment rule.
+  enforceNamedSlices?: boolean;
```

Comment:
This does more than just enforce named slices though... it changes the behavior of ordering too.  I wonder if we can come up w/ a better configuration name.  Maybe not.  I'm struggling.

Reply:
Hmm... this is tricky, since it is one option that changes two behaviors: named slices must be referenced by name, and slice ordering now reflects the rule order on the Instance. Something like `newSliceBehavior` is vague enough to cover both, but also vague enough to be basically meaningless. Maybe `manualSliceOrdering` would be okay? That name implies that the author has control over slice ordering, and if they want to control where named slices appear in the order, they would have to refer to the slice by name.

Reply:
Looking at this again, I think this option should actually go under `instanceOptions` (i.e., in the `ConfigurationInstanceOptions` type) -- since really it is an option specific to how instances are processed.  This also give the option a little more context (which helps understanding).

I think `manualSliceOrdering` could work. I also thought about `assignSlicesByName` or `requireSliceNames` -- but those still don't get at the idea that they are ordered differently.  Would `manageSlicesByName` imply ordering?  I dunno.  I think any of those will work in the context of `instanceOptions`.  Feel free to move forward with what you feel is best (and/or get input from the team).